src/fx/AnimateableTimeStretchPitchScaleReader.cpp

```diff
@@ -45,14 +45,45 @@ void AnimateableTimeStretchPitchScaleReader::read(int& length, bool& eos, sample
 
 void AnimateableTimeStretchPitchScaleReader::seek(int position)
 {
-	double time = double(position) / double(m_reader->getSpecs().rate);
+	double sampleRate = double(m_reader->getSpecs().rate);
+	double time = double(position) / sampleRate;
 	float frame = time * m_fps;
 
 	float timeRatio = m_timeStretch->readSingle(frame);
 	setTimeRatio(timeRatio);
 	float pitchScale = m_pitchScale->readSingle(frame);
 	setPitchScale(pitchScale);
-	TimeStretchPitchScaleReader::seek(position);
-}
 
+	int inputSamplePos = 0;
+	double outputSamplePos = 0.0;
+
+	float ratio = 1.0f;
+	float lastRatio = 1.0f;
+
+	const double frameDuration = 1.0 / m_fps;
+	const double samplesPerFrame = frameDuration * sampleRate;
+	int frameIndex = 0;
+
+	while(outputSamplePos < position)
```

Comment:
Would be better to change this to a for loop and you can convert the position to an integer frame count to loop over (`int(frame)`) with `frameIndex`. Also: currently you're reading too far, i.e.,  `outputSamplePos` overshoots `position`. When you cast the frame count to int, rounding down, it will instead undershoot. See a comment further below how to get to the goal more precisely.